inc/az_ulib_ustream_forward_base.h

```diff
@@ -0,0 +1,374 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#ifndef AZ_ULIB_ustream_forward_BASE_H
+#define AZ_ULIB_ustream_forward_BASE_H
+
+#include "az_ulib_config.h"
+#include "az_ulib_pal_os.h"
+#include "az_ulib_result.h"
+#include "azure/core/az_span.h"
+#include "azure/core/az_context.h"
+
+#ifdef __cplusplus
+#include <cstddef>
+#include <cstdint>
+#else
+#include <stddef.h>
+#include <stdint.h>
+#endif /* __cplusplus */
+
+#include "azure/core/_az_cfg_prefix.h"
+
+/**
+ * @brief   Define offset_t with the same size as size_t.
+ */
+typedef size_t offset_t;
```

Comment:
Because we do not expose offset, I would not create this offset_t, just use size_t for the `current_position`.

Reply:
this is brought over from ustream. I believe it is done this way as to make the "offset" type more descriptive for readability. if we change it here we should also change it for ustream. thoughts?

Reply:
I'm going to keep all things brought over from ustream identical to ustream for the first feature push